src/python/python/ad/optimizers.py

```diff
@@ -279,6 +279,13 @@ def step(self):
 
             m_tp, v_tp = self.state[k]
             m_t = self.beta_1 * m_tp + (1 - self.beta_1) * g_p
+
+            # When computing variance, want to take element-wise product of g_p
```

Comment:
Maybe write:

"Ensure that we take the element-wise product to compute variance"